app/assets/javascripts/discourse/app/form-kit/components/fk/control/checkbox.gjs

```diff
@@ -8,8 +8,8 @@ export default class FKControlCheckbox extends Component {
   static controlType = "checkbox";
 
   @action
-  handleInput() {
-    this.args.field.set(!this.args.value);
+  handleInput(event) {
+    this.args.field.set(event.target.checked);
```

Comment:
This will need to be broken out into its own PR, just keeping it here for ease of iteration for now.

Reply:
Can you clarify the problem that this fixes? Your change looks good to me, but I can't see anything wrong in old code either. Unless it's somehow possible to trigger an `input` event twice without changing the state of the checkbox?

Reply:
The issue here is that `this.args.value` always seems to be the initial value, rather than reflecting the current checked stated of the checkbox, when means that the `onSet` handler will only ever receive the same value, rather than having the value change if the checkbox is checked/unchecked multiple times.

Reply:
Okay, I figured out the problem. Setting `onSet` causes `set` to not be called on the data model, which caused the behaviour I was observing.

That said, I'd expect listening to a psuedo-event (ie, `onSet`) to not have side effects, so perhaps that behaviour should be changed? That's a much bigger change, though.

Reply:
I have pushed what I would expect in this situation: https://github.com/discourse/discourse/pull/29696/commits/539aeea4c04f044f39ec55d4cc9676dd81f45772